src/pipewire_listener.rs

```diff
@@ -52,6 +53,165 @@ enum PipewireObjectType {
     Link,
 }
 
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub enum PortConnections {
+    AllInOrder,
+    Only(Vec<(String, String)>),
+}
+
+#[derive(Debug, Clone)]
+pub struct StreamConnections {
+    pub output_stream: String,
+    pub input_stream: String,
+    pub port_connections: PortConnections,
+}
+
+#[derive(Debug)]
+pub struct StreamDescriptor {
+    pub name: String,
+    pub inputs: Vec<String>,
+    pub ouputs: Vec<String>,
+}
+
+pub struct PipewireController {
+    sender: pipewire::channel::Sender<Vec<StreamConnections>>,
+    state: Arc<Mutex<PipewireState>>,
+}
+
+impl PipewireController {
+    pub fn new() -> Self {
+        let (sender, receiver) = pipewire::channel::channel();
+        let state = Arc::new(Mutex::new(PipewireState::new()));
+        thread::spawn({
```

Comment:
tu hold pas le return value de thread::spawn(...)? When does this thread get joined()?

Reply:
non on join pas on a pas besoin ca roule jusqu'a la fin du prog